app/services/blog_service.rb

```
@@ -0,0 +1,85 @@
+    return [] unless File.exist?(MANIFEST_PATH)
+
+    @_cached_manifest_posts ||= begin
+      yaml_content = File.read(MANIFEST_PATH)
+      # Process ERB in the YAML file (for dynamic dates in our sample)
+      # In a real scenario, dates in manifest would likely be static.
+      erb_processed_yaml = ERB.new(yaml_content).result
+
+      posts_data = YAML.safe_load(erb_processed_yaml, permitted_classes: [Date, Symbol], aliases: true)
+      (posts_data || []).map do |post_hash|
+        # Convert string dates from YAML to Date objects if they are not already
+        parsed_date = post_hash['date']
```

RuboCop (1.73)

[convention] Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

(Style/StringLiterals)